.github/workflows/push-build-downstream.yaml

```diff
@@ -0,0 +1,81 @@
+name: Push Image Build Downstream
+on:
+  push:
+    branches:
+      - master
+      - release-*
+    tags:
+      - v*
+  pull_request:
+    branches:
+      - master
+
+defaults:
+  run:
+    # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+permissions:
+  contents: read
+
+jobs:
+  push-image-to-container-registry:
+    runs-on: ubuntu-20.04
+    if: github.repository == 'red-hat-storage/rook'
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "1.21"
+
+        # docker/setup-qemu action installs QEMU static binaries, which are used to run builders for architectures other than the host.
+      - name: set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: all
+
+      - name: log in to container registry
+        uses: docker/login-action@v3
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_ROBOT_TOKEN }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_USR }}
+          aws-secret-access-key: ${{ secrets.AWS_PSW }}
+          aws-region: us-east-1
+
+      # creating custom env var
+      - name: set env
+        run: |
+          echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV
+          echo "GITHUB_REF"=${GITHUB_REF} >> $GITHUB_ENV
+
+      - name: Install dependencies
+        run: |
+          python3 -m pip install --upgrade pip setuptools wheel
+          sudo apt-get -q update
+          sudo apt-get -q install -y python3-pygit2
+
+      - name: Configure git user
+        run: |
+          # Use our Rook "service account"
+          git config user.name Rook
+          git config user.email "cncf-rook-info@lists.cncf.io"
+
+      - name: build and release
```

Comment:
What tags is OCS operator expecting to find for the rook images? The publish script by default will set a tag such as `rook/ceph:v1.13.0.417.g6ab962c84`. Perhaps we should set a tag that is consistent based on the branch name such as `master` or `release-4.16`. Then the ocs operator can consistently use the latest build whenever the new image is pushed with the same tag.

Reply:
So it will auto get the branch name like this 
```
 echo BRANCH_NAME=release-4.16,           echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV

```

Reply:
for the current branch it is taking as 
```
echo BRANCH_NAME=merge
+ echo GITHUB_REF=refs/pull/594/merge
```